lib/teleterm/apiserver/handler/handler_clusters.go

```diff
@@ -76,44 +76,48 @@ func (s *Handler) RemoveCluster(ctx context.Context, req *api.RemoveClusterReque
 
 // GetCluster returns a cluster
 func (s *Handler) GetCluster(ctx context.Context, req *api.GetClusterRequest) (*api.Cluster, error) {
-	cluster, err := s.DaemonService.ResolveFullCluster(ctx, req.ClusterUri)
+	cluster, err := s.DaemonService.ResolveClusterWithDetails(ctx, req.ClusterUri)
 	if err != nil {
 		return nil, trace.Wrap(err)
 	}
 
-	return newAPIRootCluster(cluster), nil
+	return newAPIRootClusterWithDetails(cluster), nil
 }
 
 func newAPIRootCluster(cluster *clusters.Cluster) *api.Cluster {
+	loggedInUser := cluster.GetLoggedInUser()
+
 	apiCluster := &api.Cluster{
-		Uri:           cluster.URI.String(),
-		Name:          cluster.Name,
-		ProxyHost:     cluster.GetProxyHost(),
-		Connected:     cluster.Connected(),
-		AuthClusterId: cluster.AuthClusterID,
-		LoggedInUser:  newAPILoggedInUser(cluster.LoggedInUser),
+		Uri:       cluster.URI.String(),
+		Name:      cluster.Name,
+		ProxyHost: cluster.GetProxyHost(),
+		Connected: cluster.Connected(),
+		LoggedInUser: &api.LoggedInUser{
+			Name:           loggedInUser.Name,
+			SshLogins:      loggedInUser.SSHLogins,
+			Roles:          loggedInUser.Roles,
+			ActiveRequests: loggedInUser.ActiveRequests,
+		},
 	}
 
-	// Only include features in the api response if they
-	// exist on the supplied cluster
+	return apiCluster
+}
+
+func newAPIRootClusterWithDetails(cluster *clusters.ClusterWithDetails) *api.Cluster {
+	apiCluster := newAPIRootCluster(cluster.Cluster)
+
+	// Only include if they exist on the supplied cluster
 	if cluster.Features != nil {
```

Comment:
I think we can get rid of this `if` now?

Reply:
yeah seems appropriate now

Reply:
Removed